packages/driver/src/util/serialization/log.ts

```diff
@@ -327,6 +327,11 @@ export const reifyLogLikeFromSerialization = (props, matchElementsAgainstSnapsho
  * @returns a serializable form of a snapshot, including a serializable <body> with styles
  */
 export const preprocessSnapshotForSerialization = (snapshot) => {
+  // if the protocol is enabled, we don't need to preprocess the snapshot since it is serializable
+  if (Cypress.config('protocolEnabled')) {
```

Comment:
Do we need to check for `numTestsKeptInMemory` in `run` mode? Then again I don't think this actually gets called when `numTestsKeptInMemory` is zero it might be irrelevant, which makes we wonder if this actually works.

Are you saying DOM nodes are serializable over CDP?

Reply:
Yes, the `numTestsKeptInMemory` check will also be needed for the driver tests. Added [039215f](https://github.com/cypress-io/cypress/pull/26940/commits/039215f393480001fd2329d94c72774c1477a838).

> Are you saying DOM nodes are serializable over CDP?

No, the protocol snapshots only contain the name, timestamp, and highlight element selectors.